cmd/soroban-cli/src/config/network.rs

```diff
@@ -103,6 +136,21 @@ pub struct Network {
     pub network_passphrase: String,
 }
 
+fn parse_http_header(header: &str) -> Result<(String, String), Error> {
+    let header_components = header.splitn(2, ':');
+
+    let (key, value) = header_components
+        .map(str::trim)
+        .next_tuple()
+        .ok_or_else(|| Error::InvalidHeader)?;
```

Comment:
I pushed a change to trim values, and to tweak the error message. The error generated by this line looks like this now:
```
$ stellar network add lfg \
    --rpc-url http://example.org \
    --network-passphrase "Example Network" \
    --rpc-header 'Authorization'
error: invalid value 'Authorization' for '--rpc-header <RPC_HEADERS>': invalid HTTP header: must be in the form 'key:value'
```

I removed passing in the string into the error, because the clap error handler will print out the value they passed in at the start of the error message anyway.